Emergency backport of a submitted upstream patch.

From: Stefan Liebler via Libc-alpha <libc-alpha@sourceware.org>
Subject: [PATCH] s390x: Update math: redirect roundeven function
To: libc-alpha@sourceware.org
Cc: Stefan Liebler <stli@linux.ibm.com>
Date: Mon, 28 Jun 2021 13:01:07 +0200 (4 hours, 41 minutes, 28 seconds ago)
Message-Id: <20210628110107.4039211-1-stli@linux.ibm.com>
Reply-To: Stefan Liebler <stli@linux.ibm.com>

After recent commit
447954a206837b5f153869cfeeeab44631c3fac9
"math: redirect roundeven function", building on
s390x fails with:
Error: symbol `__roundevenl' is already defined

Similar to aarch64/riscv fix, this patch redirects target
specific functions for s390x:
commit 3213ed770cbc5821920d16caa93c85e92dd7b9f6
"Update math: redirect roundeven function"
---
 sysdeps/s390/fpu/s_roundeven.c  | 2 +-
 sysdeps/s390/fpu/s_roundevenf.c | 1 +
 sysdeps/s390/fpu/s_roundevenl.c | 1 +
 3 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/sysdeps/s390/fpu/s_roundeven.c b/sysdeps/s390/fpu/s_roundeven.c
index 40b07e054b..0773adfed0 100644
--- a/sysdeps/s390/fpu/s_roundeven.c
+++ b/sysdeps/s390/fpu/s_roundeven.c
@@ -18,6 +18,7 @@
    <https://www.gnu.org/licenses/>.  */
 
 #ifdef HAVE_S390_MIN_Z196_ZARCH_ASM_SUPPORT
+# define NO_MATH_REDIRECT
 # include <math.h>
 # include <libm-alias-double.h>
 
@@ -31,7 +32,6 @@ __roundeven (double x)
   __asm__ ("fidbra %0,4,%1,4" : "=f" (y) : "f" (x));
   return y;
 }
-hidden_def (__roundeven)
 libm_alias_double (__roundeven, roundeven)
 
 #else
diff --git a/sysdeps/s390/fpu/s_roundevenf.c b/sysdeps/s390/fpu/s_roundevenf.c
index d2fbf3d2b6..289785bc4a 100644
--- a/sysdeps/s390/fpu/s_roundevenf.c
+++ b/sysdeps/s390/fpu/s_roundevenf.c
@@ -18,6 +18,7 @@
    <https://www.gnu.org/licenses/>.  */
 
 #ifdef HAVE_S390_MIN_Z196_ZARCH_ASM_SUPPORT
+# define NO_MATH_REDIRECT
 # include <math.h>
 # include <libm-alias-float.h>
 
diff --git a/sysdeps/s390/fpu/s_roundevenl.c b/sysdeps/s390/fpu/s_roundevenl.c
index 29ab7a8616..94b6459ab4 100644
--- a/sysdeps/s390/fpu/s_roundevenl.c
+++ b/sysdeps/s390/fpu/s_roundevenl.c
@@ -18,6 +18,7 @@
    <https://www.gnu.org/licenses/>.  */
 
 #ifdef HAVE_S390_MIN_Z196_ZARCH_ASM_SUPPORT
+# define NO_MATH_REDIRECT
 # include <math.h>
 # include <math_private.h>
 # include <libm-alias-ldouble.h>
-- 

2.31.1

